runtime/cudaq/platform/default/rest/helpers/ionq/IonQServerHelper.cpp

```diff
@@ -0,0 +1,258 @@
+#include <fstream>
+#include <thread>
+
+namespace cudaq {
+
+/// @brief The IonQServerHelper class extends the ServerHelper class to handle
+/// interactions with the IonQ server for submitting and retrieving quantum
+/// computation jobs.
+class IonQServerHelper : public ServerHelper {
+private:
+  /// @brief RestClient used for HTTP requests.
+  RestClient client;
+
+  /// @brief Helper method to retrieve the value of an environment variable.
+  std::string getEnvVar(const std::string &key) const;
+
+  /// @brief Helper method to check if a key exists in the configuration.
+  bool keyExists(const std::string &key) const;
+
+public:
+  /// @brief Returns the name of the server helper.
+  const std::string name() const override { return "ionq"; }
+
+  /// @brief Returns the headers for the server requests.
+  RestHeaders getHeaders() override;
+
+  /// @brief Initializes the server helper with the provided backend
+  /// configuration.
+  void initialize(BackendConfig config) override;
+
+  /// @brief Creates a quantum computation job using the provided kernel
+  /// executions and returns the corresponding payload.
+  ServerJobPayload
+  createJob(std::vector<KernelExecution> &circuitCodes) override;
+
+  /// @brief Extracts the job ID from the server's response to a job submission.
+  std::string extractJobId(ServerMessage &postResponse) override;
+
+  /// @brief Constructs the URL for retrieving a job based on the server's
+  /// response to a job submission.
+  std::string constructGetJobPath(ServerMessage &postResponse) override;
+
+  /// @brief Constructs the URL for retrieving a job based on a job ID.
+  std::string constructGetJobPath(std::string &jobId) override;
+
+  /// @brief Constructs the URL for retrieving the results of a job based on the
+  /// server's response to a job submission.
+  std::string constructGetResultsPath(ServerMessage &postResponse);
+
+  /// @brief Constructs the URL for retrieving the results of a job based on a
+  /// job ID.
+  std::string constructGetResultsPath(std::string &jobId);
+
+  /// @brief Retrieves the results of a job using the provided path.
+  ServerMessage getResults(std::string &resultsGetPath);
+
+  /// @brief Checks if a job is done based on the server's response to a job
+  /// retrieval request.
+  bool jobIsDone(ServerMessage &getJobResponse) override;
+
+  /// @brief Processes the server's response to a job retrieval request and
+  /// maps the results back to sample results.
+  cudaq::sample_result processResults(ServerMessage &postJobResponse) override;
+};
+
+// Initialize the IonQ server helper with a given backend configuration
+void IonQServerHelper::initialize(BackendConfig config) {
+  cudaq::info("Initializing IonQ Backend.");
+  // Move the passed config into the member variable backendConfig
+  backendConfig = std::move(config);
+  // Set the necessary configuration variables for the IonQ API
+  backendConfig["url"] = "https://api.ionq.co";
```

Comment:
Here you actually want to take `backendConfig["url"]` and set it on a member variable for your class. This method is the framework giving you information to influence the setup of the ServerHelper, as opposed to you giving it information. This is how the Mock Test should work, notice it is setting the url to localhost, so you can grab that here and use that in your post strings.